### DISCUSSION TITLE
$RNFirebaseAnalyticsWithoutAdIdSupport = true in EAS for EXPO

### DISCUSSION THREAD

**Discussion by wortkotze** [invertase/react-native-firebase#6180] [Q&A]
Has someone an working example of an eas configuration to set 
$RNFirebaseAnalyticsWithoutAdIdSupport = true 

Where do I need to add this configuration?

thx for the help
Stephan

**Comment by mikehardy** [comment]
It needs to be set in the Podfile, here is an example of a what a Podfile would look like with it set, if you uncomment this line: https://github.com/invertase/react-native-firebase/blob/ef7895cd49eed3a4d103df624e52315250c68ab3/tests/ios/Podfile#L12

I have no idea what facilities EAS provides for adding things to Podfiles, but whatever those are, that's what you need to have as your end result when you use them

**Reply by wortkotze** [reply]
hm ok thx for the quick reply. I assume it would make sense to create a plugin for it or?
is there a reason why this is available for some of the modules but not for analytics?

**Reply by mikehardy** [reply]
I'm not sure what EAS offers, as mentioned, so it may make sense to make a plugin, it may not. There are only expo plugins for use cases that meet the combination of "a plugin is needed" and "someone has taken the time to write one". For that reason at least, firestore does not need one (does not need one) while analytics might need one but no one has taken the time to write one.

For Podfile variable inclusion there might be an easy way at least, @barthap may know a trick?

**Reply by barthap** [reply]
Expo config plugin docs mention Podfile modification here: [link](https://docs.expo.dev/guides/config-plugins/#modifying-the-ios-podfile). I haven't tested the way with `Podfile.properties.json` mentioned there, but for such a simple change, we could leverage old good "regex/codemod plugins" to modify the Podfile contents. It can be done the same way as we currently do in the RNFB/app plugin which does this:
1. Get the `Podfile` contents (even with `fs.readFileSync`, not sure if `@expo/config-plugins` has dedicated Podfile mod API)
2. Prepend it with the line `$RNFirebaseAnalyticsWithoutAdIdSupport = true\n`
3. Save the file with the new content

**Comment by safaiyeh** [comment]
addressed here! https://github.com/invertase/react-native-firebase/pull/8969

We can close this discussion when deployed.
